packages/bitpiewallet/src/adapter.ts

```diff
@@ -0,0 +1,146 @@
+import {
+    EventEmitter,
+    pollUntilReady,
+    WalletAccountError,
+    WalletAdapter,
+    WalletAdapterEvents,
+    WalletDisconnectedError,
+    WalletNotConnectedError,
+    WalletNotFoundError,
+    WalletNotInstalledError,
+    WalletPublicKeyError,
+    WalletSignatureError,
+} from '@solana/wallet-adapter-base';
+import { PublicKey, Transaction } from '@solana/web3.js';
+
+interface BitpieWallet {
+    isBitpieWallet?: boolean;
+    getAccount: () => Promise<string>;
+    signTransaction: (transaction: Transaction) => Promise<Transaction>;
+    signAllTransactions: (transactions: Transaction[]) => Promise<Transaction[]>;
+}
+
+interface BitpieWalletWindow extends Window {
+    solana?: BitpieWallet;
+}
```

Comment:
We have been asking wallets to please consider not attaching themselves to `window.solana`. This causes conflicts with other wallet extensions the user may have installed, and prevents dapps from detecting which wallets a user has.

Would you consider using `window.bitpie` or something else? We are coordinating with existing wallet providers to migrate away from using `window.solana` as well.

Reply:
Ok, i have changed to use window.bitpie

Reply:
Thank you!